universal-application-tool-0.0.1/app/repository/QuestionRepository.java

```diff
@@ -33,6 +33,57 @@ public QuestionRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext exec
         executionContext);
   }
 
+  static class PathConflictDetector {
+    Boolean conflict = Boolean.FALSE;
+    String newPath;
+
+    public PathConflictDetector(String newPath) {
+      this.newPath = checkNotNull(newPath);
+    }
+
+    public Boolean hasConflict() {
+      return conflict;
+    }
+
+    public boolean checkConflict(Question question) {
+      boolean proceed = true;
+      if (pathConflicts(question.getPath(), newPath)) {
+        conflict = Boolean.TRUE;
```

Comment:
question: why `Boolean.TRUE` rather than `true`?

Reply:
It ultimately needs to be passed to `CompletionStage<Boolean>`, we can convert the type later though. (I am not sure if `CompletionStage<boolean>` is allowed. If so we should just use that.)

Reply:
Ah I see. `CompletionStage<boolean>` isn't allowed because Java doesn't support primitives in generics.